### PR TITLE
Added TTL param to actor timer and reminder

### PR DESCRIPTION
**Pull request by amulyavarote**
Signed-off-by: Amulya Varote <amulyavarote@microsoft.com>

# Description

Added TTL param to actor timer and reminder.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #141 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Comment by XavierGeerinck** [comment]
Is there a file missing? I see a commit with a test file but I can't find it in the review?

File: daprdocs/content/en/js-sdk-docs/js-actors/_index.md [hunk 1]
```diff
@@ -184,6 +185,7 @@ async function start()
   await client.actor.reminderCreate(DemoActorImpl.name, `actor-id`, `timer-id`, {
     dueTime: Temporal.Duration.from({ seconds: 2 }),
     period: Temporal.Duration.from({ seconds: 1 }),
+    ttl?: Temporal.Duration.from({ seconds: 1 }),
```
Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/js-actors/_index.md:188]: Optional parameter should be after required parameter.

Comment (shubham1172) [inline daprdocs/content/en/js-sdk-docs/js-actors/_index.md:188]:
```suggestion
    ttl: Temporal.Duration.from({ seconds: 1 }),
```

File: src/actors/runtime/AbstractActor.ts [hunk 2]
```diff
@@ -80,14 +80,16 @@ export default abstract class AbstractActor {
    * @param reminderName name of the reminder
    * @param state the state to be send along with the reminder trigger
    * @param dueTime due time for the first trigger
+   * @param ttl time to live
```
Comment (shubham1172) [inline src/actors/runtime/AbstractActor.ts:83]:
```suggestion
   * @param ttl time duration after which the reminder will be expired and deleted
```

File: src/actors/runtime/ActorReminderData.ts [hunk 3]
```diff
@@ -20,17 +20,20 @@ export default class ActorReminderData {
...
 
   /**
    * @param reminderName the name of the actor reminder
    * @param state the state data passed to receiveReminder callback
    * @param dueTime the amount of time to delay before invoking the reminder for the first time
+   * @param ttl time to live
    * @param period the time interval between reminder invocations after the first invocation
    */
-  constructor(reminderName: string, dueTime: number, period: number, state?: string | object) {
+  constructor(reminderName: string, dueTime: number, ttl: number, period: number, state?: string | object) {
```
Comment (shubham1172) [inline src/actors/runtime/ActorReminderData.ts:33]: Same comments as above for `ttl`'s param documentation and optional nature.

File: test/actor/DemoActorReminder2Impl.ts [hunk 4]
```diff
@@ -18,7 +18,7 @@ export default class DemoActorReminder2Impl extends AbstractActor implements Dem
   counter = 0;
 
   async init(): Promise<string> {
-    await super.registerActorReminder("my-reminder-name", Temporal.Duration.from({ seconds: 2 }), Temporal.Duration.from({ seconds: 1 }), 123);
+    await super.registerActorReminder("my-reminder-name", Temporal.Duration.from({ seconds: 2 }), Temporal.Duration.from({ seconds: 2 }), Temporal.Duration.from({ seconds: 1 }), 123);
```
Comment (shubham1172) [inline test/actor/DemoActorReminder2Impl.ts:21]: nit: line break in params for readability

File: test/actor/DemoActorReminderImpl.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@ export default class DemoActorReminderImpl extends AbstractActor implements Demo
   counter = 0;
 
   async init(): Promise<string> {
-    await super.registerActorReminder("my-reminder-name", Temporal.Duration.from({ seconds: 2 }), Temporal.Duration.from({ seconds: 1 }), 123);
+    await super.registerActorReminder("my-reminder-name", Temporal.Duration.from({ seconds: 2 }), Temporal.Duration.from({ seconds: 2 }), Temporal.Duration.from({ seconds: 1 }), 123);
```
Comment (shubham1172) [inline test/actor/DemoActorReminderImpl.ts:21]: nit: line break in params for readability

**Review by shubham1172 — CHANGES_REQUESTED** [review]
LGTM @amulyavarote, only had a few nits.

**Pull request merged**